pkg/user/tx_client_test.go

```diff
@@ -208,11 +245,62 @@ func (suite *TxClientTestSuite) TestConfirmTx() {
 		msg := bank.NewMsgSend(addr, testnode.RandomAddress().(sdk.AccAddress), sdk.NewCoins(sdk.NewInt64Coin(app.BondDenom, 1+balance)))
 		resp, err := suite.txClient.BroadcastTx(suite.ctx.GoContext(), []sdk.Msg{msg}, fee, gas)
 		require.NoError(t, err)
-		require.NotNil(t, resp)
-		_, err = suite.txClient.ConfirmTx(suite.ctx.GoContext(), resp.TxHash)
+		require.Equal(t, resp.Code, abci.CodeTypeOK)
+
+		nonce, signer, exists := suite.txClient.GetTxFromLocalMempool(resp.TxHash)
+		require.True(t, exists)
+		require.Equal(t, suite.txClient.DefaultAccountName(), signer)
+		seq := suite.txClient.Signer().Account(suite.txClient.DefaultAccountName()).Sequence()
+		// Successfully broadcast transaction increases the nonce
+		// txInfo is indexed before the nonce is increased
+		require.Equal(t, seq, nonce+1)
```

Comment:
[not blocking]

looks like we repeat a few things in these test runs, is it worth using a table driven test or pulling this out?

Reply:
I can pull this out into a helper yea